proposals/0051-serialized-manifest.md

```diff
@@ -0,0 +1,171 @@
+
+## Use the manifest
+
+Internally, we already have this information, which is the `SSRManifest`, **not** the `AstroConfig`. The `AstroConfig` disappears after the build, instead we have
+the `SSRManifest` that is created during the build, and passed to the Astro application even when it has on-demand pages.
+
+The source code already does a very good job to serialise the information across pipelines (dev, SSG, SSR and container) and make sure that all of them contain the same information. Many of this information are computed from the user configuration.
+
+## What to expose
+
+As a rule of thumb, the information to expose should adhere to the following guidelines:
+- Information that can be serialised. Functions, `Map`, `Set`, etc. shouldn't be exposed. As long as a field can go through `JSON.strigify`, it's a good candidate. 
+- Information that can be used directly by our end-users. Integration developers have access to `AstroConfig`.
+- Information that isn't present inside `AstroConfig`. As mentioned previously, the `SSRManifest` contains useful information that are often used by on-demand pages and adapters.
+  An example is `SSRManifest.i18n.domainLookupTable`, which is used by adapters to understand the mapping of a domain to a locale.
+
+These are guidelines, so exceptions are possible, as long as they have a good reason to be exposed. 
+
+## Proposed APIs
+
+Proposed name of the module: `astro:manifest`
+
+Proposed sub-paths
+
+**By feature**:
+- `astro:manifest/routing`, exported bindings: 
+  - `i18n`
+  - `trailingSlash`
+  - `base`
+  - `build.format`
+  - `site` 
+  - `redirects`[^1]
+- `astro:manifest/fs`, exposed bindings: 
+  - `srcDirc`
+  - `cacheDir`
+  - `outDir`
+  - `build.client`
+  - `build.server`
+  - `build.serverEntry`
+  - `build.assetsPrefix`
+  - `publicDir`
+  - `root`
+
+**By "position"**
+- `astro:manifest/client`, exported bindings:
+  - `i18n`
+  - `trailingSlash`
+  - `base`
+  - `build.format`
+  - `site`
+  - `redirects`[^1]
+- `astro:manifest/server`, exposed bindings:
+  - `srcDirc`
+  - `cacheDir`
+  - `outDir`
+  - `build.client`
+  - `build.server`
+  - `build.serverEntry`
+  - `build.assetsPrefix`
+  - `publicDir`
+  - `root`
+
+To note that `/routing`/`/client` can be used anywhere in the code, even in client scripts.
+
+[^1]: To evaluate, probably not needed
+
+# Testing Strategy
+
+- The virtual modules will be "usable" via experimental flag. Attempting to use these virtual modules without the experimental flag turned on will result into a hard error.
+- After a gracing period where we stabilise the APIs, the experimental flag will be removed.
+- We will deprecate existing means to read this information. The following will be deprecated:
```

Comment:
I think we should also deprecate `Astro.site`

Reply:
Keep them coming, it's possible I missed some of them

Reply:
Not deprecated but that also means for actions we won't need the trailing slash placeholder

Reply:
Can this be moved to somewhere else? Mentioning deprecations belongs in Detailed design I would think, but definitely not in Testing Strategy.